src/components/header/gist-modal/renderer.tsx

```diff
@@ -132,109 +132,106 @@ class GistModal extends React.PureComponent<Props, State> {
+          }
+        });
+        if (jsonFiles.length === 0) {
+          this.setState(
+            {
+              gistLoadClicked: false,
+              invalidUrl: true
+            },
+            () => {
+              return Promise.reject('No JSON file exists in the gist');
             }
-          });
-          if (jsonFiles.length === 0) {
-            this.setState(
-              {
-                gistLoadClicked: false,
-                invalidUrl: true
-              },
-              () => {
-                return Promise.reject('No JSON file exists in the gist');
+          );
+        } else {
+          this.setState(
+            {
+              gist: {
+                ...this.state.gist,
+                filename: jsonFiles[0]
               }
-            );
-          } else {
-            this.setState(
-              {
-                gist: {
-                  ...this.state.gist,
-                  filename: jsonFiles[0]
-                }
-              },
-              () => {
-                const {revision, filename} = this.state.gist;
-                JSON.parse(json.files[jsonFiles[0]].content);
-                if (this.state.latestRevision) {
-                  this.props.history.push(`/gist/${gistId}/${filename}`);
-                } else {
-                  this.props.history.push(`/gist/${gistId}/${revision}/${filename}`);
-                }
-                closePortal();
+            },
+            () => {
+              const {revision, filename} = this.state.gist;
+              JSON.parse(gistSummary.files[jsonFiles[0]].content);
+              if (this.state.latestRevision) {
+                this.props.history.push(`/gist/${gistId}/${filename}`);
+              } else {
+                this.props.history.push(`/gist/${gistId}/${revision}/${filename}`);
               }
-            );
-          }
-        } else {
-          if (json.files[this.state.gist.filename] === undefined) {
-            this.setState({
-              gistLoadClicked: false,
-              invalidFilename: true
-            });
-            return Promise.reject('Invalid file name');
-          } else {
-            const {revision, filename} = this.state.gist;
-            JSON.parse(json.files[filename].content);
-            if (this.state.latestRevision) {
-              this.props.history.push(`/gist/${gistId}/${filename}`);
-            } else {
-              this.props.history.push(`/gist/${gistId}/${revision}/${filename}`);
+              closePortal();
             }
-            closePortal();
-          }
+          );
         }
-      })
-      .catch(error => {
-        if (error instanceof SyntaxError) {
+      } else {
+        if (gistSummary.files[this.state.gist.filename] === undefined) {
           this.setState({
             gistLoadClicked: false,
-            syntaxError: true
+            invalidFilename: true
           });
+          return Promise.reject('Invalid file name');
```

Comment:
Why are we rejecting a promise instead of raising an error?

Reply:
We are in an async function so raising an error should do the same.

Reply:
right, i forgot changing those as well.

Reply:
Let's change them before merging.

Reply:
This one was passed as callback but that didn't work properly (blank page was displayed) so I moved it out of the function call:
https://github.com/vega/editor/pull/619/commits/46754d39373990f4606052088c954aef9e1459ca#diff-d5eed843722bc0e9999038c61414bb4dL183-L185